packages/myst-transforms/src/basic.ts

```diff
@@ -19,14 +19,14 @@ export function basicTransformations(tree: GenericParent, file: VFile, opts?: Re
   // lifting roles and directives must happen before the mystTarget transformation
   liftMystDirectivesAndRolesTransform(tree);
   // Some specifics about the ordering are noted below
+  // Target transformation must happen after lifting the directives, and before the heading labels
+  mystTargetsTransform(tree, file);
```

Comment:
This rearranging is not totally necessary. However, in `getFrontmatter` we must run `mystTargetsTransform` and _all_ it's prerequisites. By moving this up earlier, we explicitly show that the only dependency is `liftMystDirectivesAndRolesTransform`.